More colourful weather icons

michaelundwd opened the pull request:
This PR updates the set of weather icons to provide greater visual clarity, especially for the 1hr interval forecasts.  It does this by a consistent use of solid colour and better vertical alignment of the elements.  There is also a new icon called drizzle that results in four changes to icon_dict in belchertown.py and belchertown.js.tmpl.  My thanks to @fvirgola80 for his helpful modifications in developing them.

To use the new icons, you will need to restart weewx after installing them so that they are copied across to your server.

You can see the icons in use at http://lordshipweather.uk and the differences from the existing ones are shown below.

New icons:
![image](https://user-images.githubusercontent.com/45148992/114269084-ac82e580-99fc-11eb-9d47-98943d5d4ef0.png)


Current icons:
![image](https://user-images.githubusercontent.com/45148992/114268998-354d5180-99fc-11eb-83f6-570c8ebd8d1b.png)

fvirgola80 commented:
🔝🔝🔝@michaelundwd 😮😉

norecords commented:
@michaelundwd do you know this [weather icon set](https://github.com/nrkno/yr-weather-symbols) ?
I try them on my skin since few weeks. That need to be reworked but I found it more light/dark theme friendly.

michaelundwd commented:
@norecords Thanks for this and yes, I did see them on your website.  It gave me the idea to making the existing ones more colourful and consistent as I do like the clean design  of the existing set.  I found there were too many snow icons and not enough rain ones in the Norwegian set you refer to and I don't like the  Aeris set that would be the obvious choice.
My next step is to move the icon_dict to an external json file, making it available to both belchertown.py and belchertown.js.tmpl.  I see no point in having two copies that have to be maintained.

roe-dl commented:
I found the current icons in their minimalist form quite beautiful. Why change them?

michaelundwd commented:
I have made two more changes to the revised image set in the latest commit, as well as some tidying up some pixels I had missed.

1. Deleted windy.png, as it does not figure in icon_dict
2. Changed tornado.png to be more illustrative

@roe-dl It is of course a matter of personal  taste about the icon design.  I did like the originals, but felt that they could be improved especially for the 1hr interval forecasts.
It is for @pob to decide whether to merge them into the master or not, depending on the feedback he gets.  All I have done here is to provide an alternative.  It may be possible for alternative sets to be available. 

Revised icon set below

![image](https://user-images.githubusercontent.com/45148992/114445309-d4727480-9bc7-11eb-869f-7695f44a3b92.png)

Overcon commented:
How do we apply these to our website?

michaelundwd commented:
You download the latest icons from the  PR https://github.com/poblabs/weewx-belchertown/pull/667/commits/02ac5f7672f2079adda6723c3553829c7cfd8355 and copy them to the images folder in your belchertown root folder.  You then restart weewx so they get copied to the images folder in your html_root folder.

Alternatively they are in the coloured_icons.zip folder below.  Unzip them and then follow the instructions as above.

[coloured_icons.zip](https://github.com/poblabs/weewx-belchertown/files/6415671/coloured_icons.zip)

Overcon commented:
Very cool, thank you!

ddperso commented:
Very nice, I like

poblabs commented:
@michaelundwd Couple of conflicts in this PR now due to its age - sorry! I'll see if I can resolve it. 

My question - who created these icons? We should give credit in the readme.md

michaelundwd commented:
@pob - good to hear from you.

The conflict arises because this PR was done before #672, and the merge has committed #672 first.  #672 had the icon_dict embedded in belchertown.py and belchertown.js.tmpl; #672 moved them to a common external file.  So the new master code incorporating #672 Does that help, or do you want me to do more?  

I created these icons with the help of @fvirgola80.  They are a further modification of the icons that I amended in #388, #389.  So the originals came from "Brian at weather34.com for the weather icons from the simplicty 2015 theme. Used with agreement." as mentioned in your credits.  I have modified and extended the set, ie more remixing that you have already acknowledged. 
Thanks
Michael

poblabs commented:
For this and #649 I could use your help determining how to resolve the conflict. If possible, you can re-sync your fork of the project, update the files and commit to your branch. Then each PR wil have  the right code and the conflicts should go away and I can merge

michaelundwd commented:
ok, will do for this and #649.  I'll resync to the latest master and see what happens!
